graphql/client.go

```diff
@@ -61,13 +61,17 @@ type client struct {
 //
 // The typical method of adding authentication headers is to wrap the client's
 // Transport to add those headers.  See example/caller.go for an example.
-func NewClient(endpoint string, httpClient *http.Client) Client {
+func NewClient(endpoint string, httpClient Doer) Client {
 	if httpClient == nil {
```

Comment:
To be safe we should also now check `|| httpClient == (*http.Client)(nil)`, in case someone does e.g.
```go
var httpClient *http.Client
if false { httpClient = ... }
NewClient("...", httpClient)
```

Reply:
Good point!